lib/index.js

```diff
@@ -82,18 +86,24 @@ Optimizely.prototype.initialize = function() {
  */
 
 Optimizely.prototype.track = function(track) {
-  var payload = {};
+  // Use the new-style API (which is compatible with Classic and X)
+  var payload = {
+    type: 'event',
+    eventName: track.event(),
+    tags: track.properties()
+  };
   var revenue = track.revenue();
-  if (revenue) payload.revenue = revenue *= 100;
-  push(['trackEvent', track.event(), payload]);
+  if (revenue) payload.tags.revenue *= 100;
+
+  push(payload);
 
   var optimizelyClientInstance = window.optimizelyClientInstance;
   if (optimizelyClientInstance && optimizelyClientInstance.track) {
     var attributes = track.properties();
     var optimizelyOptions = track.options('Optimizely');
     var userId = optimizelyOptions.userId || track.userId();
     if (userId) {
-      optimizelyClientInstance.track(track.event(), userId, attributes, payload.revenue);
+      optimizelyClientInstance.track(track.event(), userId, attributes, payload.tags.revenue);
```

Comment:
`payload.tags.revenue` would be `undefined` here if `properties.revenue` doesn't exist. Is that okay? Also `attributes` being passed here would include `attributes.revenue` since line 102 is pulling all properties.